### ISSUE TITLE
Circle from 2 tangents map tool works only if snapping to segment is checked

### ISSUE THREAD

**Issue by 3nids** [qgis/QGIS#46703]
### What is the bug or the crash?

Circle from 2 tangents map tool works only if snapping to segment is checked
I would expected the tool to work without any specific snapping config.

### Steps to reproduce the issue

use the map tool

### Versions

master

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

@lbartoletti is this expected?

**Comment by lbartoletti** [comment]
> @lbartoletti is this expected?

Yes it was expected. But, I agree with you, this tool should work without any specific snapping config.

**Comment by lbartoletti** [comment]
BTW, if we fix this tool, we should also fix trim/extend and circle from 3 tangents.
